swarm/api/config.go

```diff
@@ -104,23 +104,35 @@ func NewConfig() (c *Config) {
 
 //some config params need to be initialized after the complete
 //config building phase is completed (e.g. due to overriding flags)
-func (c *Config) Init(prvKey *ecdsa.PrivateKey) {
+func (c *Config) Init(prvKey *ecdsa.PrivateKey) error {
```

Comment:
After changing the interface - by adding `error` to return - please make sure that test also handle the error coming from `Init()`.